packages/fxa-content-server/app/scripts/views/mixins/do-not-sync-mixin.js

```diff
@@ -7,14 +7,14 @@
  *
  * @mixin DoNotSync mixin
  */
-import SigninMixin from './signin-mixin';
 import FlowEventsMixin from './flow-events-mixin';
+import preventDefaultThen from '../decorators/prevent_default_then';
 
 export default {
-  dependsOn: [SigninMixin, FlowEventsMixin],
+  dependsOn: [FlowEventsMixin],
 
   events: {
-    'click #do-not-sync-device': 'doNotSync',
+    'click #do-not-sync-device': preventDefaultThen('doNotSync'),
```

Comment:
`preventDefaultThen` is needed to prevent the browser from pulling the user back to /choose_what_to_sync after the form submit on the /confirm_signup_code screen. Strange.